crates/ef-testing/src/sequencer/utils.rs

```diff
@@ -0,0 +1,92 @@
+use blockifier::abi::abi_utils::get_storage_var_address as blockifier_get_storage_var_address;
+use reth_primitives::{Bytes, TransactionSigned};
+use reth_rlp::Decodable;
+use revm_primitives::U256;
+use starknet::{
+    core::types::{BroadcastedInvokeTransaction, FieldElement},
+    macros::selector,
+};
+use starknet_api::{
+    core::{ClassHash, ContractAddress},
+    hash::StarkFelt,
+    state::StorageKey,
+};
+
+use super::{constants::KAKAROT_ADDRESS, KakarotSequencer};
+
+pub(crate) fn get_storage_var_address(
+    storage_var: &str,
+    keys: &[StarkFelt],
+) -> Result<StorageKey, eyre::Error> {
+    Ok(blockifier_get_storage_var_address(storage_var, keys)?)
+}
+
+pub(crate) fn split_bytecode_to_starkfelt(bytecode: &Bytes) -> Vec<StarkFelt> {
+    bytecode
+        .chunks(16)
+        .map(|x| {
+            let mut storage_value = [0u8; 16];
+            storage_value[..x.len()].copy_from_slice(x);
+            StarkFelt::from(u128::from_be_bytes(storage_value))
+        })
+        .collect()
+}
+
+pub(crate) fn split_u256(value: U256) -> [u128; 2] {
+    [
+        (value & U256::from(u128::MAX)).try_into().unwrap(), // safe unwrap <= U128::MAX.
+        (value >> 128).try_into().unwrap(),                  // safe unwrap <= U128::MAX.
+    ]
+}
+
+pub(crate) fn contract_address_to_starkfelt(contract_address: &ContractAddress) -> StarkFelt {
```

Comment:
in the `types.rs` you defined a very similar `TryFrom<FeltSequencer> for ContractAddress`, + I'm wondering why we should do this since these two types are from starknet_api

Reply:
Sometimes I need to get the Felt inside the contract address, I can remove this and replace everywhere with `contract_address.0.key()`, just find it nicer like this

Reply:
Could implement `From` for it yes

Reply:
I'm worried the PR will take a lot of time to get merged. I can make one but last merged PR on starknet api was 3 weeks ago